runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -6229,8 +6234,8 @@ int chpl_comm_addr_gettable(c_nodeid_t node, void* start, size_t len)
             return;                                                     \
           }                                                             \
                                                                         \
-          if (IS_32_BIT_AMO_ON_GEMINI(_t)                               \
-              || mreg_for_remote_addr(obj, loc) == NULL) {              \
+          remote_mr = mreg_for_remote_addr(obj, loc);                   \
+          if (IS_32_BIT_AMO_ON_GEMINI(_t) || remote_mr == NULL) {       \
```

Comment:
Truly a nit, and apologies for not catching this in my first quick look, but the old code was written with the `||` the way it was specifically so that for 32-bit AMOs on Gemini we didn't do a remote MR lookup.  (There was no need for that lookup because Gemini doesn't support 32-bit AMOs and thus those have to be done by AMs.)  Of course we could recover that old behavior by recoding as:

```
if (IS_32_BIT_AMO_ON_GEMINI(_t)
    || (remote_mr = mreg_for_remote_addr(obj, loc)) == NULL) {
...
```
XE performance is certainly secondary at this point, but at the same time if we're willing to go to some effort to avoid small overheads on XC we probably ought to do so on other platforms as well, even old ones.

Reply:
Right, I considered noting that in the PR message. I was going to say something like "This optimizes the common/fast path, but it will actually result in an unnecessary remote memory lookup for 32 bit ops on gemini"

I can still note that, but I don't think we care that much because XE isn't that important to us anymore, and most atomics operations are going to be on 64-bit stuff anyways.

Reply:
I think there are 7 more of these below, where the MR lookup call has now been hoisted above the 32-bit Gemini check that would render it superfluous.

Reply:
Most AMOs are indeed 64-bit, but absent a pressing need (which textual simplification isn't) I don't see any reason to reduce current XE performance.  And Blue Waters and Titan are still running, so Chapel could end up there.

Reply:
Yeah, good point. I'll make that mod